examples/subscribing_to_stream.rs

```diff
@@ -0,0 +1,112 @@
+use eventstore::{
+    Client, Credentials, EventData, ExpectedVersion, Origin, Position, ReadResult, SubEvent,
+    SubscribeToAllOptions, SubscribeToStreamOptions, SubscriptionFilter,
+};
+use futures::TryStreamExt;
+use serde::{Deserialize, Serialize};
+use std::error::Error;
+use uuid::Uuid;
+
+type Result<A> = std::result::Result<A, Box<dyn Error>>;
+
+pub async fn subscribe_to_stream(client: &Client) -> Result<()> {
+    // region subscribe-to-stream
+    client
+        .subscribe_to_stream("some-stream", &Default::default())
+        .await?;
+    // endregion subscribe-to-stream
+
+    // region subscribe-to-stream-from-position
+    let options = SubscribeToStreamOptions::default().origin(Origin::Point(20));
+
+    client.subscribe_to_stream("some-stream", &options).await?;
+    // endregion subscribe-to-stream-from-position
+
+    // region subscribe-to-stream-live
+    let options = SubscribeToStreamOptions::default().origin(Origin::End);
+    client.subscribe_to_stream("some-stream", &options).await?;
+    // endregion subscribe-to-stream-live
+
+    // region subscribe-to-stream-resolving-linktos
+    let options = SubscribeToStreamOptions::default()
+        .origin(Origin::Start)
+        .resolve_links();
+
+    client
+        .subscribe_to_stream("$et-myEventType", &options)
+        .await?;
+    // endregion subscribe-to-stream-resolving-linktos
+
+    // region subscribe-to-stream-subscription-dropped
+    let mut stream = client
```

Comment:
**FIX:** This sample should show handling when subscription connection was dropped. I might not be understanding this code well enough, but I don't see here any error handling.

Reply:
If the `while` loop completes then it means the subscription has dropped.

Reply:
I understand, but it's not explicit - see that other clients show where you should do some reconnect logic, e.g. https://github.com/EventStore/EventStoreDB-Client-Java/blob/trunk/db-client-java/src/test/java/com/eventstore/dbclient/samples/subscribing_to_stream/SubscribingToStream.java#L50. Can we have some way of marking it here?

Reply:
We agree we should show the resubscribing strategy.

Reply:
done